osu.Game/Screens/SelectV2/PanelBeatmapSet.cs

```diff
@@ -164,5 +174,30 @@ protected override void FreeAfterUse()
             updateButton.BeatmapSet = null;
             difficultiesDisplay.BeatmapSet = null;
         }
+
+        public override MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                if (Item == null)
+                    return Array.Empty<MenuItem>();
+
+                var beatmapSet = (BeatmapSetInfo)Item.Model;
+
+                List<MenuItem> items = new List<MenuItem>();
+
+                if (!Expanded.Value)
+                {
+                    items.Add(new OsuMenuItem("Expand", MenuItemType.Highlighted, () => TriggerClick()));
+                    items.Add(new OsuMenuItemSpacer());
+                }
+
+                if (beatmapSet.OnlineID > 0)
+                    items.Add(new OsuMenuItem("Details...", MenuItemType.Standard, () => beatmapOverlay?.FetchAndShowBeatmapSet(beatmapSet.OnlineID)));
+
+                items.Add(new OsuMenuItem("Delete...", MenuItemType.Destructive, () => songSelect?.Delete(beatmapSet)));
+                return items.ToArray();
```

Comment:
Compared to [v1 implementation](https://github.com/ppy/osu/blob/faa73f5d04fdc6be1fe3d8921bbc793523e7eb0b/osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs#L280-L308), is missing:

- collections management (being able to add the set to a collection, pulling up manage dialog)
- restore all hidden
- copy link to clipboard

Reply:
i'll look at the latter two. for collections see the note in OP